### PR TITLE
chore(lint): Some more rules enforcing consistency

### PR DESCRIPTION
**Pull request by vhf**
Here are a few more rules I feel are missing and would improve consistency.

### CODE REVIEW

File: static/.eslintrc [hunk 1]
```diff
@@ -41,18 +41,39 @@ Do not edit this file in your local project
     "comma-dangle": [2, "never"],
     "comma-spacing": [2, { "before": false, "after": true }],
     "comma-style": [2, "last"],
+    "computed-property-spacing": [2, "never"],
```
Comment (marcbachmann) [inline static/.eslintrc]: Does `standard/computed-property-even-spacing` only cover newlines or didn't we install it properly?
https://github.com/upfrontIO/apply-guides/pull/27/files#diff-2072934b5106b60a8a6ffc4737a34a21R194

We depend on `eslint-plugin-standard`, which isn't installed in the editor.

File: static/.eslintrc [hunk 2]
```diff
@@ -133,6 +166,7 @@ Do not edit this file in your local project
     "no-useless-rename": 2,
     "no-whitespace-before-property": 2,
     "no-with": 2,
+    "object-curly-spacing": [2, "always"],
```
Comment (marcbachmann) [inline static/.eslintrc]: Please keep it at `never`
https://github.com/upfrontIO/livingdocs/blob/master/contribution-guidelines/style-guides/coffeescript.md#our-rules

File: static/.eslintrc [hunk 3]
```diff
@@ -87,12 +108,24 @@ Do not edit this file in your local project
...
+      ],
+      "allowSamePrecedence": false
+    }],
     "no-mixed-spaces-and-tabs": 2,
     "no-multi-spaces": 2,
     "no-multi-str": 2,
     "no-multiple-empty-lines": [2, { "max": 2 }],
     "no-native-reassign": 2,
     "no-negated-in-lhs": 2,
+    "no-nested-ternary": 2,
```
Comment (marcbachmann) [inline static/.eslintrc]: 👍

File: static/.eslintrc [hunk 4]
```diff
@@ -87,12 +108,24 @@ Do not edit this file in your local project
     "no-label-var": 2,
     "no-labels": [2, { "allowLoop": false, "allowSwitch": false }],
     "no-lone-blocks": 2,
+    "no-lonely-if": 2,
+    "no-mixed-operators": [2, {
```
Comment (marcbachmann) [inline static/.eslintrc]: 👍

File: static/.eslintrc [hunk 5]
```diff
@@ -87,12 +108,24 @@ Do not edit this file in your local project
...
+      "groups": [
```
Comment (marcbachmann) [inline static/.eslintrc]: > Default is the groups for each kind of operators.

I guess you can remove that property. That's used by default.

File: static/.eslintrc [hunk 6]
```diff
@@ -87,12 +108,24 @@ Do not edit this file in your local project
...
+        ["+", "-", "*", "/", "%", "**"],
+        ["&", "|", "^", "~", "<<", ">>", ">>>"],
+        ["==", "!=", "===", "!==", ">", ">=", "<", "<="],
+        ["&&", "||"],
+        ["in", "instanceof"]
+      ],
+      "allowSamePrecedence": false
```
Comment (marcbachmann) [inline static/.eslintrc]: This isn't very reasonable for me.
e.g. `var foo = a + b - c;` wouldn't be allowed

http://eslint.org/docs/rules/no-mixed-operators#allowsameprecedence

File: static/.eslintrc [hunk 7]
```diff
@@ -87,12 +108,24 @@ Do not edit this file in your local project
     "no-label-var": 2,
     "no-labels": [2, { "allowLoop": false, "allowSwitch": false }],
     "no-lone-blocks": 2,
+    "no-lonely-if": 2,
```
Comment (marcbachmann) [inline static/.eslintrc]: http://eslint.org/docs/rules/no-lonely-if#rule-details

I'm using the second invalid example quite often
```js
if (condition) {
    // ...
} else {
    if (anotherCondition) {
        // ...
    } else {
        // ...
    }
}
```

Do you think that's hard to read?

and that would be valid.
```js
if (condition) {
    // ...
} else {
    if (anotherCondition) {
        // ...
    }
    doSomething();
}
```

IMO This rule can cause confusion if somebody needs to modify some existing code.

File: static/.eslintrc [hunk 8]
```diff
@@ -41,18 +41,39 @@ Do not edit this file in your local project
     "comma-dangle": [2, "never"],
     "comma-spacing": [2, { "before": false, "after": true }],
     "comma-style": [2, "last"],
+    "computed-property-spacing": [2, "never"],
     "constructor-super": 2,
     "curly": [2, "multi-line"],
     "dot-location": [2, "property"],
-    "eol-last": 2,
+    "eol-last": [2, "always"],
```
Comment (marcbachmann) [inline static/.eslintrc]: 🎉 This wasn't enforceable in coffee-script.
But isn't `always` the default?

File: static/.eslintrc [hunk 9]
```diff
@@ -41,18 +41,39 @@ Do not edit this file in your local project
...
+    "linebreak-style": [2, "unix"],
+    "max-len": ["error", {
+      "code": 100,
+      "ignoreRegExpLiterals": true,
+      "ignorePattern": "^\\s*const\\s.+=\\s*require\\s*\\(/"
+    }],
+    "multiline-ternary": ["off", "never"],
     "new-cap": [2, { "newIsCap": true, "capIsNew": false }],
     "new-parens": 2,
+    "newline-per-chained-call": [2, { "ignoreChainWithDepth": 4 }],
```
Comment (marcbachmann) [inline static/.eslintrc]: I guess that's fine. Do you think 3 is too low?

File: static/.eslintrc [hunk 10]
```diff
@@ -41,18 +41,39 @@ Do not edit this file in your local project
...
+    }],
     "key-spacing": [2, { "beforeColon": false, "afterColon": true }],
     "keyword-spacing": [2, { "before": true, "after": true }],
+    "linebreak-style": [2, "unix"],
+    "max-len": ["error", {
+      "code": 100,
+      "ignoreRegExpLiterals": true,
+      "ignorePattern": "^\\s*const\\s.+=\\s*require\\s*\\(/"
+    }],
+    "multiline-ternary": ["off", "never"],
```
Comment (marcbachmann) [inline static/.eslintrc]: Hmm. they are more readable than on one line.
IMO prevent nested ternaries is good enough.

**Review by marcbachmann — CHANGES_REQUESTED** [review]

File: static/.eslintrc (hunk 1 above)
Comment (vhf) [reply static/.eslintrc]: It depends whether we really want consistent style or not. `even` only enforces you put N spaces after `[` and before `]`. N can be 0, 1, ...

I'd rather have a strict rule saying `foo[bar]` so N = 0.

Comment (marcbachmann) [reply static/.eslintrc]: Yes, that's better 👍 
Can we remove the other config then?

**Comment by vhf** [comment]
@masone this one is ready, we discussed each rule together with @marcbachmann and @peyerluk .

**Comment by vhf** [comment]
Should I fix `index.js` by the way?

**Comment by marcbachmann** [comment]
> Should I fix index.js by the way?

Yes, please. Thanks.

**Comment by marcbachmann** [comment]
Our bot is happy :grin:

**Comment by masone** [comment]
Thanks for pushing forward on this! I like!

**Comment by marcbachmann** [comment]
@masone you just made comment confusing by deleting the one of the bot -.-

**Review by marcbachmann — APPROVED** [review]

**Pull request merged**